sql/src/test/java/io/crate/integrationtests/JoinIntegrationTest.java

```diff
@@ -291,6 +291,17 @@ public void testCrossJoinSysTablesOnly() throws Exception {
                "2| 0| t\n"));
     }
 
+    @Test
+    public void testJoinUsingSubscriptInQuerySpec() {
+        execute("create table t1 (id int, a object as (b int))");
+        execute("create table t2 (id int)");
+        execute("insert into t1 (id, a) values (1, {b=1})");
+        execute("insert into t2 (id) values (1)");
+        refresh();
+        execute("select t.id, tt.id from t1 as t, t2 as tt where tt.id = t.a['b']");
+        assertThat(printedTable(response.rows()), is("1| 1\n"));
```

Comment:
not sure if we wanna keep, the planner test already validates correct plans but something could still be wrong when building/executing the the plan

Reply:
I'd say given the subscript complexity we keep this as well